qt/scientific_interfaces/Indirect/IndirectFitAnalysisTab.cpp

```diff
@@ -16,347 +128,648 @@ namespace IDA {
+/**
+ * @return  The function index of the selected background.
  */
-void IndirectFitAnalysisTab::setDefaultPropertyValue(
-    const QString &propertyName, const double &propertyValue) {
-  m_defaultPropertyValues[propertyName] = propertyValue;
+int IndirectFitAnalysisTab::backgroundIndex() const {
+  return m_fitPropertyBrowser->backgroundIndex();
+}
 
-  if (m_propertyFunctions.contains(propertyName))
-    updateProperty(propertyName, selectedSpectrum());
+/**
+ * @return  The fit type selected in the custom functions combo box, in the fit
+ *          property browser.
+ */
+QString IndirectFitAnalysisTab::selectedFitType() const {
+  return m_fitPropertyBrowser->selectedFitType();
 }
 
-/*
- * Removes the default value for the property with the specified name,
- * in the property table of this fit analysis tab.
+/**
+ * @param functionName  The name of the function.
+ * @return              The number of custom functions, with the specified name,
+ *                      included in the selected model.
+ */
+size_t IndirectFitAnalysisTab::numberOfCustomFunctions(
+    const std::string &functionName) const {
+  return m_fitPropertyBrowser->numberOfCustomFunctions(functionName);
+}
+
+/**
+ * @return  The selected Start-X value in the indirect fit analysis tab.
+ */
+double IndirectFitAnalysisTab::startX() const {
+  return m_fitPropertyBrowser->startX();
+}
+
+/**
+ * @return  The selected End-X value in the indirect fit analysis tab.
+ */
+double IndirectFitAnalysisTab::endX() const {
+  return m_fitPropertyBrowser->endX();
+}
+
+/**
+ * @param functionName  The name of the function containing the parameter.
+ * @param parameterName The name of the parameter whose value to retrieve.
+ * @return              The value of the parameter with the specified name, in
+ *                      the function with the specified name.
+ */
+double
+IndirectFitAnalysisTab::parameterValue(const std::string &functionName,
+                                       const std::string &parameterName) {
+  return m_fitPropertyBrowser->parameterValue(functionName, parameterName);
+}
+
+/**
+ * @return  True if the selected model is empty, false otherwise.
+ */
+bool IndirectFitAnalysisTab::emptyModel() const {
+  auto modelFunction = model();
+  auto compositeModel =
+      boost::dynamic_pointer_cast<CompositeFunction>(modelFunction);
+
+  if (compositeModel)
+    return compositeModel->nFunctions() == 0;
+  else
+    return modelFunction->asString() == "";
+}
+
+/**
+ * @return  The name of the selected background.
+ */
+QString IndirectFitAnalysisTab::backgroundName() const {
+  return m_fitPropertyBrowser->backgroundName();
+}
+
+/**
+ * @return  True if the current model is the same as the model which was most
+ *          recently fit, false otherwise.
+ */
+bool IndirectFitAnalysisTab::previousFitModelSelected() const {
+  return equivalentFunctions(m_fitFunction,
+                             m_fitPropertyBrowser->compositeFunction());
+}
```

Comment:
Why do you keep your fit function in two different places?

Reply:
m_fitFunction is the function which was previously used in a fit.
m_fitPropertyBrowser->compositeFunction() returns the currently selected function in the fit property browser.